pkg/api/v1beta3/dynakube/extensions.go

```diff
@@ -0,0 +1,104 @@
+	OpenTelemetryCollector OpenTelemetryCollectorSpec `json:"openTelemetryCollector,omitempty"`
+}
+
+type PrometheusSpec struct {
+	Enabled bool `json:"enabled"`
+}
+
+type ExtensionExecutionControllerSpec struct {
+
+	// Defines storage device
+	// +kubebuilder:validation:Optional
+	PersistentVolumeClaim *corev1.PersistentVolumeClaimSpec `json:"persistentVolumeClaim,omitempty"`
+
+	// Adds additional labels for the ExtensionExecutionController pods
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Adds additional annotations to the ExtensionExecutionController pods
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Overrides the default image
+	// +kubebuilder:validation:Optional
+	ImageRef ImageRefSpec `json:"imageRef,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	TlsRefName string `json:"tlsRefName,omitempty"`
+
+	// Define resources' requests and limits for single ExtensionExecutionController pod
+	// +kubebuilder:validation:Optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Determines retention policy
+	// +kubebuilder:validation:Optional
+	PersistentVolumeClaimRetentionPolicy *appsv1.PersistentVolumeClaimRetentionPolicyType `json:"persistentVolumeClaimRetentionPolicy,omitempty"`
+
+	// Set tolerations for the ExtensionExecutionController pods
+	// +kubebuilder:validation:Optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Adds TopologySpreadConstraints for the ExtensionExecutionController pods
+	// +kubebuilder:validation:Optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+}
+
+type OpenTelemetryCollectorSpec struct {
+
+	// Adds additional labels for the OtelCollector pods
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Adds additional annotations to the OtelCollector pods
+	// +kubebuilder:validation:Optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Overrides the default image
+	// +kubebuilder:validation:Optional
+	ImageRef ImageRefSpec `json:"imageRef,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	TlsRefName string `json:"tlsRefName,omitempty"`
+
+	// Define resources' requests and limits for single OtelCollector pod
+	// +kubebuilder:validation:Optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Set tolerations for the OtelCollector pods
+	// +kubebuilder:validation:Optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Adds TopologySpreadConstraints for the OtelCollector pods
+	// +kubebuilder:validation:Optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+
+	// Number of replicas for your OtelCollector
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	Replicas int32 `json:"replicas,omitempty"`
```

Comment:
general rule of thumb, don't use `omitempty` and `kubebuilder:default` at the same time, because then you won't be able to set a value to its "default-golang-value"

Reply:
`omitempty` removed. The value can be set to 0. The property has the default value if not provided by a user.